python/graphframes/lib/pregel.py

```diff
@@ -16,13 +16,17 @@
 #
 
 import sys
-from typing import Any
-if sys.version > '3':
+from typing import TYPE_CHECKING, Any
+
+if sys.version > "3":
     basestring = str
 
+from pyspark.ml.wrapper import JavaWrapper
 from pyspark.sql import DataFrame, SparkSession
 from pyspark.sql.functions import col
-from pyspark.ml.wrapper import JavaWrapper
+
+if TYPE_CHECKING:
+    from graphframes import GraphFrame
```

Comment:
is this right?

Reply:
Yes. It is needed in the only line and only for type hints. `TYPE_CHECKING` variable is exactly for that case when we need something only during the static analysis, but not in the runtime.

Reply:
https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING